src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt

```diff
@@ -57,29 +92,41 @@ class PestFinder {
     fun onPestSpawn(event: PestSpawnEvent) {
         if (!isEnabled()) return
         PestSpawnTimer.lastSpawnTime = SimpleTimeMark.now()
-        val plot = GardenPlotAPI.getPlotByName(event.plotName)
-        if (plot == null) {
-            ChatUtils.userError("Open Desk to load plot names and pest locations!")
-            return
+        val plotNames = event.plotNames
+        for (plotName in plotNames) {
+            val plot = getPlotByName(plotName)
+            if (plot == null) {
+                ChatUtils.userError("Open Plot Management Menu to load plot names and pest locations!")
+                return
+            }
+            if (event.unkownAmount) {
+                plot.isPestCountInaccurate = true
+            } else {
+                plot.pests += event.amountPests
+                plot.isPestCountInaccurate = false
+            }
         }
-        plot.pests += event.amountPests
         update()
     }
 
     @SubscribeEvent
     fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
-        if (!isEnabled()) return
+        if (!GardenAPI.inGarden()) return
         if (event.inventoryName != "Configure Plots") return
 
         for (plot in GardenPlotAPI.plots) {
-            plot.pests = 0
-            val item = event.inventoryItems[plot.inventorySlot] ?: continue
-            for (line in item.getLore()) {
-                pestInventoryPattern.matchMatcher(line) {
-                    plot.pests = group("amount").formatNumber().toInt()
+            if (!plot.isBarn()) {
```

Comment:
this cant be an early return i think, since the reason this check is here is to not accidentally set the amount of pests in the barn plot as 0, as hypixel doesnt show the amount of pests in it in the plots menu

Reply:
continue, not return. just switch the if statement. intellij should suggest you the correct action already

Reply:
weird intellij wasnt suggesting anything